src/figuresClasses.ts

```diff
@@ -1,11 +1,86 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: string;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if (
+      sideA >= sideB + sideC ||
+      sideB >= sideA + sideC ||
+      sideC >= sideA + sideB
+    ) {
+      throw new Error(
+        'The sum of the lengths of any two sides ' +
+          'must be greater than the length of the remaining side.',
+      );
+    }
+
+    if (sideA <= 0 || sideB <= 0 || sideC <= 0) {
+      throw new Error('All sides must be greater than 0.');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.sideA + this.sideB + this.sideC) / 2;
+    const area = Math.sqrt(
+      s * (s - this.sideA) * (s - this.sideB) * (s - this.sideC),
+    );
+
+    return Math.floor(area * 100) / 100;
```

Comment:
Consider returning the exact area value instead of rounding it within the `getArea` method. This would allow the consumer of the method to decide how to handle the precision of the value.